docs/hacking.rst

```diff
@@ -2,273 +2,256 @@
 Hacking
 =======
 
-L20n library is a combination of three layers of APIs built on top of each other.
-The intention is for the user to be able to use just the first or first and the second
-layer separately.
+The L20n library is built as a combination of three layers of APIs stacked on 
+top of each other.  This modularization allows l20n.js to support many 
+different environments and platforms.
 
-Each layer has different external dependencies and is intended to tie better into the
-host and client environments.
+:ECMA 402 Intl
+    The low-level API providing L20n's primitives and building-blocks.
 
-Intl API
-========
+:Localization
+    The main l20n.js class responsible for language negotiation, resource 
+    loading and the fallback strategy.
+
+:LocalizationObserver
+    The manager class used to bind to the host environment.
+
+Each layer has different external dependencies and is intended to tie better 
+into the host and client environments.  In addition to the above three layers 
+there's also the so-called runtime code which is responsible for instantiating 
+the proper classes, attaching event listeners and triggering the initial 
+translation of the UI.
+
+
+The Intl API
+============
+
+At the heart of l20n.js is the low-level ECMA 402's Intl API.  L20n.js uses 
+``Intl.NumberFormat`` to format numbers in a locale-aware manner and 
+``Intl.DateTimeFormat`` to format dates.  Other features are provided by our custom
+extensions to the Intl API, modeled after the existing ECMA specifications:
```

Comment:
Instead, I'd suggest to put here a note that states that we intend to work with ECMA on making all APIs on this layer be aligned with ECMA402.